src/installing-workflow/configuring-registry.md

```diff
@@ -0,0 +1,56 @@
+# Configuring Registry
+
+Deis Workflow's builder component relies on registry for storing the application docker images.
+
+Deis Workflow ships with [registry][registry] component by default, which provides in-cluster docker registry backed by the platform configured [object storage][storage]. Few operators might want to use off-cluster registry mainly for performance or security reasons.
+
+## Configuring off-cluster Private Registry
+
+Every component that relies on registry uses two inputs for configuration:
+
+1. Registry Location environment variable named `DEIS_REGISTRY_LOCATION`
+2. Access credentials stored as a Kubernetes secret named `registry-secret`
+
+The helm classic chart for Deis Workflow can be easily configured to connect Workflow components to off-cluster registry. Deis Workflow currently supports Google Container Registry, Amazon ECR and other off-cluster storage providers like dockerhub, quay.io, etc., which use long lived authentication details. If using Google Container Registry or Amazon ECR, Deis Workflow will generate short lived token for authenticating with the registry such that the deployed apps will only have access to the short lived tokens and not to the actual credentials.
+
+* **Step 1:** If you haven't already fetched the Helm Classic chart, do so with `helmc fetch deis/workflow-v2.3.0`
+* **Step 2:** Update database connection details either by setting the appropriate environment variables _or_ by modifying the template file `tpl/generate_params.toml`. Note that environment variables take precedence over settings in `tpl/generate_params.toml`.
+    * **1.** Using environment variables:
+        * Set `REGISTRY_LOCATION` to `off-cluster`, `ecr` or `gcr`, then set the following environment variables accordingly.
+
+          * For `REGISTRY_LOCATION=gcr`:
+
+            ```
+            export GCR_KEY_JSON, GCR_HOSTNAME
+            ```
+
+            `GCR_HOSTNAME` can be empty and needs to be set if host name is different from "https://gcr.io"
+
+          * For `REGISTRY_LOCATION=ecr`:
+
+            ```
+            export ECR_ACCESS_KEY, ECR_SECRET_KEY, ECR_REGION, ECR_REGISTRY_ID, ECR_HOSTNAME
+            ```
+
+              * To use [IAM credentials][aws-iam], it is not necessary to set `AWS_ACCESS_KEY` or `AWS_SECRET_KEY`. If `ECR_REGISTRY_ID` is left empty then default registry of the account will be used. `ECR_HOSTNAME` needs to be if the default hostname is aliased or the cluster is behind a proxy.
+
+
+          * For `REGISTRY_LOCATION=off-cluster`:
+
+            ```
+            export REGISTRY_USERNAME, REGISTRY_PASSWORD, REGISTRY_HOSTNAME, REGISTRY_ORGANIZATION
```

Comment:
I would say yes there is no support currently as i haven't tested that scenario...From the looks of the code i will say it might work as i am not bailing out anywhere if username or password is empty.